### PR TITLE
Compute overall dice score

### PR DESCRIPTION
**Pull request by AjeyPaiK**
The PR makes changes:
1. To the way model predictions are accessed in Lightning 2.0
2. To the way how overall dice is computed during validation
3. Adds files to gitignore so that some private files may be maintained locally.

### CODE REVIEW

File: ahcore/lit_module.py [hunk 1]
```diff
@@ -303,9 +303,9 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
         gathered_preds = self.all_gather(preds)
         return gathered_preds
 
-    def on_predict_epoch_end(self, results) -> None:
+    def on_predict_epoch_end(self):
         """Call all the inference trackers to update"""
```
Comment (jonasteuwen) [inline ahcore/lit_module.py:307]: Should do -> None

File: ahcore/trackers.py [hunk 2]
```diff
@@ -36,12 +43,12 @@ def __call__(self, predictions: list[list[Tensor]], metadata: dict[str, Any], *a
...
-        compression: str | None = "jpeg",
-        quality: int | None = 100,
-        is_mask: bool = True,
+            self,
+            save_dir: Union[str, Path],
+            pyramid: bool = False,
+            compression: str | None = "jpeg",
+            quality: int | None = 100,
+            is_mask: bool = True,
     ):
```
Comment (jonasteuwen) [inline ahcore/trackers.py:52]: What happened?

**Review by jonasteuwen — COMMENTED** [review]
Can you explain a bit what you precisely changed in each file?

File: ahcore/lit_module.py (hunk 1 above)
Comment (AjeyPaiK) [reply ahcore/lit_module.py:307]: Yes, done

**Comment by AjeyPaiK** [comment]
Changes:

1. Lightning has changed the way it handles prediction outputs. Attached the screenshot below. Earlier, the model outputs were automatically consumed by `on_predict_epoch_end()` but this behavior has changed. Now, we access them through the trainer object as shown below on the right.
image
So, we make this change in [lit_module](https://github.com/NKI-AI/ahcore/pull/19/files#diff-7503b04ebd2fd6defdbb818002acf81e2ec020d0edb62137f7de35229d56e470), 

2. For my purposes, I need to maintain additional functionalities like a tracker that can measure tumor-stroma ratio during inference which doesn't align with general ahcore functionalities. So, I added such files to gitignore. This will allow me to have them in my local repo whereas the public code neednt have it. This is reflected [here](https://github.com/NKI-AI/ahcore/pull/19/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947).

3. One main change that this PR makes is it adds a flag to the way we compute dice scores over the WSIs. Instead of averaging individual dice scores over the total images, we now aggregate true_positives and cardinalities across the whole validation dataset and treat them as though they originate from a single image. Thus avoiding the need to average while providing more stable DIce estimates. This is reflected [here](https://github.com/NKI-AI/ahcore/pull/19/files#diff-c94e1ef8a11cccfb65a65f64b32c0fbfe082571843caf2c6c9a49953a38880e8).

4. Finally, I have refactored the trackers.py a little bit. I moved the tile iterator out of the Tracker class since it was not dependent on the class variables to function.

**Pull request closed without merging**